src/react-app-env.d.ts

```diff
@@ -1 +1,23 @@
 /// <reference types="react-scripts" />
+
+interface User {
+  name: string;
+  email: string;
+}
+
+interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+  user: User | null;
+  comment: Comment[] | null;
```

Comment:
It's array, so you should use plural naming for this
```suggestion
  comments: Comment[] | null;
```